### PR TITLE
[Text] Deprecate `heading2xl` and `heading3xl` variants

### PR DESCRIPTION
**Pull request by lgriffee**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris-internal/issues/1378

### WHAT is this pull request doing?
-  Adds new functionality to console.warn() for deprecated prop values
  Screenshot 2024-02-21 at 10 23 19 AM
-  Updates deprecation guide info

### CODE REVIEW

File: polaris-react/DEPRECATED.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
+# Deprecated
```
Comment (lgriffee) [inline polaris-react/DEPRECATED.md]: I don't love that this file would essentially be a duplication of our migration guide...but it does surface what we have in progress to those consuming the library (who can't easily find the next major branch and guide). Open to other ideas of what should be in this file.

File: polaris-react/src/components/Text/Text.tsx [hunk 2]
```diff
@@ -88,6 +88,19 @@ export const Text = ({
...
+  const deprecatedVariants = ['heading2xl', 'heading3xl'];
+
+  if (
+    process.env.NODE_ENV === 'development' &&
+    variant &&
+    deprecatedVariants.includes(variant)
+  ) {
+    // eslint-disable-next-line no-console
+    console.warn(
+      `Deprecation: <Text variant="${variant}" />. The value "${variant}" will be removed in a future major version of Polaris. Use "displayXl" instead.`,
```
Comment (lgriffee) [inline polaris-react/src/components/Text/Text.tsx]: I just noticed that we don't point to the DEPRECATED.md file anywhere, should this warning be where we do that?

File: polaris-react/src/components/Text/Text.tsx [hunk 3]
```diff
@@ -88,6 +88,19 @@ export const Text = ({
   visuallyHidden = false,
   textDecorationLine,
 }: TextProps) => {
+  const deprecatedVariants = ['heading2xl', 'heading3xl'];
```
Comment (aaronccasanova) [inline polaris-react/src/components/Text/Text.tsx]: Can we move this outside the function body?

File: polaris-react/src/components/Text/Text.tsx (hunk 2 above)
Comment (aaronccasanova) [inline polaris-react/src/components/Text/Text.tsx]: I'm not familiar with `displayXl`, however, I'm curious if the suggestion works for all `deprecatedVariants`? Should the `deprecatedVariants` be an array of objects with `from`/`to` properties?

Comment (aaronccasanova) [reply polaris-react/src/components/Text/Text.tsx]: Discussed offline and will be restructuring `deprecatedVariants` similar to our migration from/to replacement maps

File: polaris-react/src/components/Text/Text.tsx (hunk 3 above)
Comment (lgriffee) [reply polaris-react/src/components/Text/Text.tsx]: Updated in 9b24a61

**Review by aaronccasanova — APPROVED** [review]

**Pull request merged**